spec/unit/intercom/client_collection_proxy_spec.rb

```diff
@@ -32,4 +32,49 @@
     client.expects(:get).with("/users", {:tag_name => 'Taggart J'}).returns(page_of_users(false))
     client.users.find_all(:tag_name => 'Taggart J').map(&:email).must_equal %W(user1@example.com user2@example.com user3@example.com)
   end
+
+  it "supports single page pagination" do
+    users = [test_user("user1@example.com"), test_user("user2@example.com"), test_user("user3@example.com"),
+     test_user("user4@example.com"), test_user("user5@example.com"), test_user("user6@example.com"),
+     test_user("user7@example.com"), test_user("user8@example.com"), test_user("user9@example.com"),
+     test_user("user10@example.com")]
+    client.expects(:get).with("/users", {:type=>'users', :per_page => 10, :page => 1}).returns(users_pagination(false, per_page=10, page=1, total_pages=1, total_count=10, user_list=users))
+    result = client.users.find_all(:type=>'users', :per_page => 10, :page => 1).map {|user| user.email }
+    result.must_equal %W(user1@example.com user2@example.com user3@example.com user4@example.com user5@example.com user6@example.com user7@example.com user8@example.com user9@example.com user10@example.com)
+  end
+
+  it "supports multi page pagination" do
+    users = [test_user("user3@example.com"), test_user("user4@example.com")]
+    client.expects(:get).with("/users", {:type=>'users', :per_page => 2, :page => 3}).returns(users_pagination(true, per_page=2, page=3, total_pages=5, total_count=10, user_list=users))
+    result = client.users.find_all(:type=>'users', :per_page => 2, :page => 3).map {|user| user.email }
+    result.must_equal %W(user3@example.com user4@example.com)
+  end
+
+  it "works with page out of range request" do
+    users = []
+    client.expects(:get).with("/users", {:type=>'users', :per_page => 2, :page => 30}).returns(users_pagination(true, per_page=2, page=30, total_pages=2, total_count=3, user_list=users))
+    result = client.users.find_all(:type=>'users', :per_page => 2, :page => 30).map {|user| user.email }
+    result.must_equal %W()
+  end
+
+  it "works with asc order" do
+    test_date=1457337600
+    time_increment=1000
+    users = [test_user_dates(email="user1@example.com", created_at=test_date), test_user_dates(email="user2@example.com", created_at=test_date-time_increment),
+             test_user_dates(email="user3@example.com", created_at=test_date-2*time_increment), test_user_dates(email="user4@example.com", created_at=test_date-3*time_increment)]
+    client.expects(:get).with("/users", {:type=>'users', :per_page => 2, :page => 30, :order => "asc", :sort => "created_at"}).returns(users_pagination(true, per_page=2, page=30, total_pages=2, total_count=3, user_list=users))
+    result = client.users.find_all(:type=>'users', :per_page => 2, :page => 30, :order => "asc", :sort => "created_at").map {|user| user.email }
```

Comment:
`result = client.users.find_all(:type=>'users', :per_page => 2, :page => 30, :order => "asc", :sort => "created_at").map(&:email)`
should work as well